kedro/io/core.py

```diff
@@ -263,7 +263,7 @@ def _to_str(obj, is_root=False):
                 text = ", ".join(
                     fmt.format(key, _to_str(value))  # 2
                     for key, value in sorted_dict
-                    if value or isinstance(value, bool)
+                    if value != {} and value is not None
```

Comment:
Should we restrict this just to `is not None`? Empty dictionaries and sets should also be provided, since in Python it is not recommended to have emtpy lists, objects or sets as default values in constructors.

Reply:
I think that makes sense. I modified it to follow whatever is written in the description (didn't know if there was particular design decisions around excluding empty dictionaries--and you would know this best--but I'd figure it had to do with load and save args getting replaced with default dictionaries. Even then, I think what you say makes sense.

Right now:

```python
>>> from kedro.extras.datasets import pandas
>>> pandas.CSVDataSet(filepath="test.txt")
<kedro.extras.datasets.pandas.csv_dataset.CSVDataSet object at 0x10a9cf150>
>>> str(pandas.CSVDataSet(filepath="test.txt"))
"CSVDataSet(filepath=test.txt, protocol=file, save_args={'index': False})"
```

Your suggestion:

```python
>>> str(pandas.CSVDataSet(filepath="test.txt"))
"CSVDataSet(filepath=test.txt, protocol=file, load_args={}, save_args={'index': False})"
```

Alternative (don't process defaults; I think this may have been the intention of the above code, but I still think the outputs make less sense/are less explicit):

```python
>>> str(pandas.CSVDataSet(filepath="test.txt"))
"CSVDataSet(filepath=test.txt, protocol=file)"  # Because no args were passed and it's `None` before the default substitution.
```

Reply:
@idanov There will be some (trivial) changes to the tests to support this, but the failures also illustrate what the difference would look like; e.g. `SQLDataSet` will include `load_args={}` in the representation, `PartitionedDataSet` will have `dataset_config={}`. Just want to confirm this change is desirable first. :)